File: leetcode_py/Integer to Roman.py
class Solution:
    # @param {integer} num
    # @return {string}
    def intToRoman(self, num):
        symbol = ["M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"]
        value = [1000, 900, 500, 400, 100, 90,  50, 40,  10, 9,   5,  4,   1]
        res = ''
        while num:
            for idx, item in enumerate(value):
                if num/item != 0:
                    res += num/item * symbol[idx]
                    num %= item
        return res

class Solution:
    # @param {integer} num
    # @return {string}
    def intToRoman(self, num):
        symbol=["M","CM","D","CD","C","XC","L","XL","X","IX","V","IV","I"]
        value=[1000,900,500,400, 100, 90,  50, 40,  10, 9,   5,  4,   1]
        res=''
        while num>0:
            for idx,i in enumerate(value):
                while num/i>0:
                    res+=symbol[idx]
                    num-=i
        return res

File: leetcode_js/Integer to Roman.js
/**
 * @param {number} num
 * @return {string}
 */
var intToRoman = function(num) {
    var symbol = ["M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"];
    var value = [1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1];
    var res = '';
    while(num > 0){
    	for(var i = 0; i < value.length; i++){
    		while(num >= value[i]){
    			res += symbol[i];
    			num -= value[i];
    		}
    		// more geek
    		// var n = Math.floor(num/value[i]); 
    		// if(n != 0){
    		// 	res += new Array(n + 1).join(symbol[i]);
    		// 	num %= value[i];
    		// }
    	}
    }
    return res;
};